core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -554,6 +555,32 @@ public String authority() {
     }
   }
 
+  private final class DirectFallbackExecutor implements Executor {
+    @Override
+    public void execute(final Runnable command) {
+      final boolean[] accepted = new boolean[1];
+      try {
+        ManagedChannelImpl.this.executor.execute(new Runnable() {
+          @Override
+          public void run() {
+            accepted[0] = true;
+            command.run();
+          }
+        });
+      } catch (RejectedExecutionException e) {
```

Comment:
In that case, we can check `terminated` directly, but still a race.
If we check `executor =! null`, and run with it, there could be a NPE because of the race.